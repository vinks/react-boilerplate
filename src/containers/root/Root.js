import React from 'react'
import PropTypes from 'prop-types'
import { Provider } from 'react-redux'
import getRoutes from 'routes'
import { Router } from 'react-router'

function Root({ store, history }) {
  const routes = getRoutes(store)

  return (
    <Provider store={store}>
      <Router history={history} routes={routes} />
    </Provider>
  )
}

Root.propTypes = {
  store: PropTypes.object.isRequired,
  history: PropTypes.object.isRequired
}

export default Root
