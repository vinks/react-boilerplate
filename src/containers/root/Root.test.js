/* eslint-env jest */

jest.mock('react-redux')
jest.mock('react-router')
jest.mock('routes', () => 'routes')

import React from 'react'
import renderer from 'react-test-renderer'
import Root from 'containers/root/Root'

describe('containers/root/Root', () => {
  it('renders correctly', () => {
    const component = renderer.create(
      <Root
        store={{ store: 'store' }}
        history={{ history: 'history' }}
      />
    )

    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
