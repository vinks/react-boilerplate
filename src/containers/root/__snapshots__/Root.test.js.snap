// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/root/Root renders correctly 1`] = `
<Provider
  store={
    Object {
      "store": "store",
    }
  }
>
  <Router
    history={
      Object {
        "history": "history",
      }
    }
    routes="routes"
  />
</Provider>
`;
